### PR TITLE
Allow HEAD requests on public files

### PR DESCRIPTION
**Pull request by agnivade**
Fixes https://github.com/mattermost/mattermost/issues/25666

```release-note
NONE
```

### CODE REVIEW

File: server/channels/api4/file.go [hunk 1]
```diff
@@ -41,7 +41,7 @@ func (api *API) InitFile() {
 
 	api.BaseRoutes.Team.Handle("/files/search", api.APISessionRequiredDisableWhenBusy(searchFiles)).Methods("POST")
 
-	api.BaseRoutes.PublicFile.Handle("", api.APIHandler(getPublicFile)).Methods("GET")
+	api.BaseRoutes.PublicFile.Handle("", api.APIHandler(getPublicFile)).Methods("GET", "HEAD")
```
Comment (jupenur) [inline server/channels/api4/file.go:44]: If it's a HEAD request we shouldn't serve a response body, right?

Comment (agnivade) [reply server/channels/api4/file.go:44]: We don't. Go takes care of it.

Comment (agnivade) [reply server/channels/api4/file.go:44]: To expand, `http.ServeContent` checks the HTTP method and then only responds with the headers. https://cs.opensource.google/go/go/+/refs/tags/go1.21.6:src/net/http/fs.go;l=352-355

**Review by jupenur — APPROVED** [review]

**Pull request merged**